pkg/symbolizer/symbolizer.go

```diff
@@ -234,16 +291,31 @@ func (s *Symbolizer) Symbolize(ctx context.Context, locations []*pb.Location) er
 		}
 		logger := log.With(s.logger, "buildid", mapping.BuildId)
 
+		var liner liner
 		locations := locationsByMapping.Locations
-		level.Debug(logger).Log("msg", "storage symbolization request started", "build_id_length", len(mapping.BuildId))
 		// Symbolize returns a list of lines per location passed to it.
-		locationsByMapping.LocationsLines, err = s.symbolizeLocationsForMapping(ctx, mapping, locations)
+		locationsByMapping.LocationsLines, liner, err = s.symbolizeLocationsForMapping(ctx, mapping, locations)
 		if err != nil {
 			level.Debug(logger).Log("msg", "storage symbolization request failed", "err", err)
 			continue
 		}
-		level.Debug(logger).Log("msg", "storage symbolization request done")
+		if liner != nil {
+			newLinerCache[mapping.BuildId] = liner
+		}
 	}
+	for k := range newLinerCache {
+		delete(s.linerCache, k)
+	}
+	for _, liner := range s.linerCache {
+		// These are liners that didn't show up in the latest iteration.
```

Comment:
How do you feel about adding a metric to track the number of times we see the same build id? It could help us to fine-tune the lifecycle of a cached liner. I guess the cache is to correlate it with cache hits/misses and symbolization rounds.